pkg/util/mount/mount.go

```diff
@@ -166,34 +198,56 @@ func (mounter *SafeFormatAndMount) FormatAndMount(source string, target string,
 	return mounter.formatAndMount(source, target, fstype, options)
 }
 
-// GetMountRefsByDev finds all references to the device provided
-// by mountPath; returns a list of paths.
-func GetMountRefsByDev(mounter Interface, mountPath string) ([]string, error) {
-	mps, err := mounter.List()
+// GetMountRefsBySource finds all references to the mount source, returns a
+// list of paths. Source could be a mountpoint path, device or a normal
+// directory (for bind mount).
+func GetMountRefsBySource(mounter Interface, source string) ([]string, error) {
+	mis, err := mounter.ListMountInfos()
 	if err != nil {
 		return nil, err
 	}
-	slTarget, err := filepath.EvalSymlinks(mountPath)
+	slTarget, err := filepath.EvalSymlinks(source)
```

Comment:
I think this EvalSymlinks won't work for the same reason too.

Reply:
@msau42 I proposed a solution here: https://github.com/kubernetes/kubernetes/pull/61489#discussion_r180301910